plugins/outputs/file/file.go

```diff
@@ -13,24 +13,58 @@ import (
 	"github.com/influxdata/telegraf/internal/rotate"
 	"github.com/influxdata/telegraf/plugins/outputs"
 	"github.com/influxdata/telegraf/plugins/serializers"
+	"github.com/klauspost/compress/zstd"
 )
 
 //go:embed sample.conf
 var sampleConfig string
 
+var ValidCompressionAlgorithmLevels = map[string][]int{
+	"zstd": {1, 3, 7, 11},
+}
+
 type File struct {
 	Files               []string        `toml:"files"`
 	RotationInterval    config.Duration `toml:"rotation_interval"`
 	RotationMaxSize     config.Size     `toml:"rotation_max_size"`
 	RotationMaxArchives int             `toml:"rotation_max_archives"`
 	UseBatchFormat      bool            `toml:"use_batch_format"`
+	Compression         Compression     `toml:"compression"`
 	Log                 telegraf.Logger `toml:"-"`
 
 	writer     io.Writer
 	closers    []io.Closer
 	serializer serializers.Serializer
 }
 
+type Compression struct {
+	Enabled   bool   `toml:"enabled"`
+	Algorithm string `toml:"algorithm"`
+	Level     int    `toml:"level"`
+}
+
+func ValidateCompressionAlgorithm(algorithm string) error {
+	for validAlgorithm := range ValidCompressionAlgorithmLevels {
+		if algorithm == validAlgorithm {
+			return nil
+		}
+	}
+	return fmt.Errorf("unknown or unsupported algorithm provided: %s", algorithm)
+}
+
+func ValidateCompressionLevel(algorithm string, level int) error {
+	for _, validAlgorithmLevel := range ValidCompressionAlgorithmLevels[algorithm] {
+		if level == validAlgorithmLevel {
+			return nil
+		}
+	}
```

Comment:
We have internal functions to support this, see `choice.Check`.

Reply:
The function `choice.Check` checks for a slice of strings but I'm using integers and it won't save that many lines so I think we can keep it as it's.